### DISCUSSION TITLE
Not willing to mod my Deluge, but eager to debug with log prints

### DISCUSSION THREAD

**Discussion by soymonitus** [SynthstromAudible/DelugeFirmware#23] [Q&A]
I would not like to mod my Deluge, but still would want to somehow debug what I am developing. I saw in the docs mentioned that there was some kind of log printing happening on the MIDI port. Is it possible or will be possible soon to print logs through the MIDI port and get some kind of Windows app (or terminal app) to read them and print them? This would make debugging without modding possible, as by adding log prints in the code that are sent through MIDI (maybe sysex messages?) we can have visibility of what is going on

**Comment by alex-reid** [comment]
From what I gathered from the Q&A, this would probably be the best starting point. I'm primarily a frontend developer with a bit of experience working with midi on MCUs. Off the top of my head it should be relatively simple to create a [electron](https://www.electronjs.org/) app to pretty print any debug info from sysex messages.

**Reply by alex-reid** [reply]
extra simple proof of concept 
<img width="802" alt="image" src="https://github.com/SynthstromAudible/DelugeFirmware/assets/2162898/793fc7c9-e34d-4b45-bab8-3a5f76ad8b5c">

**Comment by m-m-adams** [comment]
Debug prints can be done via UART out of the midi port. They'll probably show up in a midi reader if you have every possible message type enabled, but they'd be pretty meaningless. The simplest method might be to get a USB UART device and then connect it with a midi breakout adapter

**Comment by m-m-adams** [comment]
Also to enable UART you set HAVE_RTT to 0 in the preprocessor options

**Comment by litui** [comment]
While I support this idea (if only because serial over midi seems pretty cool ;) ) it does need to be said that logging output is just a fractional part of what a debugger can provide. Access to registers, memory, and code execution flow are the big components that midi serial just isn't going to be able to replicate.

**Comment by designerfuzzi** [comment]
still the best tool to filter, analyse whats going on https://www.snoize.com/midimonitor/
has also a github home https://github.com/krevis/MIDIApps

**Comment by m-m-adams** [comment]
Completed via dbt midi logging or Bfredl's web app
